core/nodejsActionBase/runner.js

```diff
@@ -67,8 +67,7 @@ function NodeActionRunner() {
         } else {
             // The code is a plain old JS file.
             try {
-                eval(message.code);
-                thisRunner.userScriptMain = eval(message.main);
+                thisRunner.userScriptMain = eval('(function(){\n' + message.code + '\nreturn ' + message.main + '})()');
```

Comment:
Yes use same approach as zip, mktemp for now in tmpfs on random folder name. 
I think just using action.js as file is good and simple instead of trying to make sure action name is safe to use as filename and encoding. 

In terms of modules.exports.main I think it's fine and low risk or use exports.$message.main